pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/MissingEqualsOnComparableRule.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule.errorprone;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAnonymousClassDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTClassDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTRecordDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTTypeDeclaration;
+import net.sourceforge.pmd.lang.java.ast.internal.JavaAstUtils;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRulechainRule;
+import net.sourceforge.pmd.lang.java.types.TypeTestUtil;
+
+/**
+ * Rule that checks {@link Comparable} classes for proper equals/hashCode implementations.
+ * <p>
+ * <a href="https://www.jetbrains.com/help/inspectopedia/ComparableImplementedButEqualsNotOverridden.html">ComparableImplementedButEqualsNotOverridden</a>
+ *
+ * @author <a href="mailto:vpotucek@me.com">Vincent Rudolf Potuček</a>
+ * @since 7.16.0
+ */
+public class MissingEqualsOnComparableRule extends AbstractJavaRulechainRule {
```

Comment:
```suggestion
public class MissingEqualsOnComparableRule extends OverrideBothEqualsAndHashcodeRule {
```
this does not compile but would somehow be nice. Somehow we should reference it.


What if the other rule is not included?

Then the equals contract is still broken, despite passing this rule.

Reply:
as they would not collide anyways, we could then check for hashCode as well to really make sure and not suppress potential harm.

Or should we leave this burden to the user?

Reply:
How do you know, "they would not collide anyways"?

I was trying to say: Implementing the missing feature in PMD "You have enabled rule X, but it is recommend to enable rule Y additionally" is out of scope for this PR, so we have to make a trade-off to just document the relationship between the rules and leave this to the user.

Reply:
Sorry, but your test is not testing the correct thing. See my comment https://github.com/pmd/pmd/pull/5856/files#r2215349797